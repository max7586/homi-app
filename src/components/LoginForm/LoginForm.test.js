import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import LoginForm from './LoginForm';
import { BrowserRouter } from 'react-router-dom'

describe('LoginForm component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <BrowserRouter>
        <LoginForm />
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<LoginForm onSuccess={()=>{}}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders the UI as expected with no props', () => {
    const tree = renderer
      .create(<LoginForm />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});