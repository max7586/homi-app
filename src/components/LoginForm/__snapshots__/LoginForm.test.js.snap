// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders the UI as expected 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="user_name"
  >
    <label
      htmlFor="LoginForm__user_name"
    >
      User name
    </label>
    <input
      className="Input "
      id="LoginForm__user_name"
      name="user_name"
      required={true}
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="LoginForm__password"
    >
      Password
    </label>
    <input
      className="Input "
      id="LoginForm__password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <button
    className="Button "
    style={
      Object {
        "marginBottom": 70,
      }
    }
    type="submit"
  >
    Login
  </button>
</form>
`;

exports[`LoginForm component renders the UI as expected with no props 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="user_name"
  >
    <label
      htmlFor="LoginForm__user_name"
    >
      User name
    </label>
    <input
      className="Input "
      id="LoginForm__user_name"
      name="user_name"
      required={true}
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="LoginForm__password"
    >
      Password
    </label>
    <input
      className="Input "
      id="LoginForm__password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <button
    className="Button "
    style={
      Object {
        "marginBottom": 70,
      }
    }
    type="submit"
  >
    Login
  </button>
</form>
`;
